### PR TITLE
Delete submodule via UI

### PR DESCRIPTION
**Pull request by codingtwinky**
I was working on #513 and got side tracked and did it for submodule first....

Will do same thing for remotes in similar fashion.

screen shot 2015-03-22 at 2 00 57 pm

### CODE REVIEW

**Comment by codingtwinky** [comment]
I thought I added test but it's missing so will update it later...

File: components/submodules/submodules.js [hunk 1]
```diff
@@ -83,3 +85,23 @@ SubmodulesViewModel.prototype.submoduleLinkClick = function(submodule) {
...
+  diag.closed.add(function() {
+    if (diag.result()) {
+      self.addProgressBar.start();
+      self.server.del('/submodules', { path: self.repoPath, submodulePath: submodule.path, submoduleName: submodule.name }, function(err, result) {
+        if (err) {
+          console.log(err);
+          return;
+        }
+
+        programEvents.dispatch({ event: 'submodule-added' });
```
Comment (campersau) [inline components/submodules/submodules.js]: I think the event should be `submodule-removed`. You can also add it here: https://github.com/codingtwinky/ungit/blob/513/components/submodules/submodules.js#L21

**Comment by codingtwinky** [comment]
@campersau  "submodule-added" is an event that really doesn't suit it's name as all it does is fetch.  I just changed the name to "submodule-fetch" so it is more fitting.

**Comment by FredrikNoren** [comment]
GJ!

**Pull request merged**